libs/core/src/main/java/org/opensearch/core/common/io/stream/StreamInput.java

```diff
@@ -1305,6 +1305,21 @@ public <E extends Enum<E>> EnumSet<E> readEnumSet(Class<E> enumClass) throws IOE
         return res;
     }
 
+    /**
+     * Reads an optional enum with type E that was serialized based on the value of its ordinal
+     * The set is expected to have been written using {@link StreamOutput#writeOptionalEnumSet(EnumSet)}
+     *
+     * @return the enum set of strings
+     * @throws IOException if an I/O exception occurs reading the set
+     */
+    public <E extends Enum<E>> EnumSet<E> readOptionalEnumSet(Class<E> enumClass) throws IOException {
+        if (readBoolean()) {
+            return readEnumSet(enumClass);
+        } else {
+            return null;
```

Comment:
Looking at the [ML Commons implementation](https://github.com/opensearch-project/ml-commons/blob/d4ed7f54560c2f3a30e57c223aaaa879a2e371a8/plugin/src/main/java/org/opensearch/ml/stats/MLStatsInput.java#L119-L128), the default on false is an empty set.
```suggestion
            return EnumSet.noneOf(enumClass);
```